diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SingleLineStatementsRule.kt

```diff
@@ -44,8 +47,13 @@ class SingleLineStatementsRule : Rule("statement") {
                         if (it.treeParent.elementType == ENUM_ENTRY){
                             node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node.treeNext)
                         } else {
-                            if (!it.isBeginByNewline())
-                                node.addChild(PsiWhiteSpaceImpl("\n"), it)
+                            if (!it.isBeginByNewline()) {
+                                val nextNode = it.treeNext ?: it.parent({ parent -> parent.treeNext != null})
```

Comment:
```suggestion
                                val nextNode = it.parent({ parent -> parent.treeNext != null}, strict=false)?.treeNext
```
Shouldn't it be like this, when you call `treeNext` on parent too?